src/core/memory/memory.cc

```diff
@@ -15,13 +15,14 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#ifndef DISABLE_WARNINGS
 
+#ifdef USE_CUDA
```

Comment:
but the macro check is before include "singa/core/memory.h"..
I think we should include memory.h before macro.